src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -557,18 +562,22 @@ def _format_linux_fx_version(custom_image_name, container_config_type=None):
     return fx_version
 
 
-def _add_linux_fx_version(cmd, resource_group_name, name, custom_image_name, slot=None):
-    fx_version = _format_linux_fx_version(custom_image_name)
-    return update_site_configs(cmd, resource_group_name, name, linux_fx_version=fx_version, slot=slot)
+def _add_fx_version(cmd, resource_group_name, name, custom_image_name, slot=None):
+    fx_version = _format_fx_version(custom_image_name)
+    web_app = get_webapp(cmd, resource_group_name, name, slot)
+    linux_fx = fx_version if web_app.reserved else None
+    windows_fx = fx_version if web_app.is_xenon else None
+    return update_site_configs(cmd, resource_group_name, name,
+                               linux_fx_version=linux_fx, windows_fx_version=windows_fx, slot=slot)
 
 
 def _delete_linux_fx_version(cmd, resource_group_name, name, slot=None):
     return update_site_configs(cmd, resource_group_name, name, linux_fx_version=' ', slot=slot)
 
 
-def _get_linux_fx_version(cmd, resource_group_name, name, slot=None):
+def _get_fx_version(cmd, resource_group_name, name, slot=None):
     site_config = get_site_configs(cmd, resource_group_name, name, slot)
-    return site_config.linux_fx_version
+    return site_config.linux_fx_version if site_config.linux_fx_version else site_config.windows_fx_version
```

Comment:
it can be simplified `site_config.linux_fx_version or site_config.windows_fx_version`